### PR TITLE
Remove redundant user/client ID parameters

### PR DESCRIPTION
**Pull request by kkohbrok**

### CODE REVIEW

File: src/spec/authentication_service.md [hunk 1]
```diff
@@ -154,12 +152,11 @@ Allows users to update their [user profile](./glossary.md#user-profile). The pro
 
 ```rust
 struct UpdateUserProfileParams {
-  user_id: UserId,
   new_user_profile: Vec<u8>,
 }
 ```
 
-The AS overwrites the existing user profile ciphertext with the new one.
+The AS overwrites the existing user profile ciphertext with the new one. The target user's user ID is derived from the client's ClientCredential.
```
Comment (raphaelrobert) [inline src/spec/authentication_service.md:159]:
```suggestion
The AS overwrites the existing user profile ciphertext with the new one. The target user's ID is derived from the client's ClientCredential.
```

File: src/spec/authentication_service.md [hunk 2]
```diff
@@ -230,14 +226,13 @@ Delete the user account with the given user id.
...
 struct DeleteUserParams {
-  user_id: UserId,
   opaque_ke3: OpaqueKe3,
 }
 ```
 
 The AS performs the following actions:
 
-* look up the OPAQUE `server_state` in the ephemeral DB based on the `client_id`
+* look up the OPAQUE `server_state` in the ephemeral DB based on the `client_id` given as part of the client's credential-based authentication
```
Comment (raphaelrobert) [inline src/spec/authentication_service.md:235]: Is 'given' a synonym for 'assigned'? If so, substitute it. If not, I don't understand what it means.

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Thanks, just a few nits

File: src/spec/authentication_service.md (hunk 2 above)
Comment (kkohbrok) [reply src/spec/authentication_service.md:235]: The user account deletion endpoint requires Client 2FA authentication, for which one factor is client credential auth. What I was trying to say here is that the client id used for the look up is meant to be taken from the credential used during authentication.

Comment (raphaelrobert) [reply src/spec/authentication_service.md:235]: This just sounds unusual, "given" is not really a technical term. Do you mean "look up the OPAQUE `server_state` in the ephemeral DB based on the `client_id` assigned through the client's credential-based authentication"?

The same should also be applied to line 138.

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Just one nit

**Pull request merged**